File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> visited = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (visited.containsKey(target - nums[i])) {
                return new int[]{i, visited.get(target - nums[i])};
            }
            visited.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {}

        for i, val in enumerate(nums):
            if (target - val) in visited:
                return [visited[target - val], i]
            
            visited[val] = i
        
        